docker-compose.yml

```diff
@@ -0,0 +1,138 @@
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+version: "3.6"
+services:
+  elastic:
+    image: ghcr.io/baking-bad/bcdhub-elastic:master
+    restart: always
+    volumes:
+      - esdata:/usr/share/elasticsearch/data
+    environment:
+      - bootstrap.memory_lock=true
+      - discovery.type=single-node
+      - "ES_JAVA_OPTS=-Xms256m -Xmx256m"
+    logging: &bcd-logging
+      options:
+        max-size: 10m
+        max-file: "5"
+
+  db:
+    image: postgres:12
+    shm_size: 1g
+    restart: always
+    environment:
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=root
+      - POSTGRES_DB=indexer
+      - PGDATA=/var/lib/postgresql/data/pgdata
+    volumes:
+      - db:/var/lib/postgresql/data
+    logging: *bcd-logging
+
+  api:
+    restart: always
+    image: ghcr.io/baking-bad/bcdhub-api:master
+    environment:
+      - BCD_ENV=sandbox
+      - GIN_MODE=debug
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=root
+      - SANDBOX_NODE_URI=http://sandbox:20000
+      - SANDBOX_IPFS_GATEWAY=https://cloudflare-ipfs.com
+    depends_on:
+      - elastic
+      - db
+    ports:
+      - 127.0.0.1:14000:14000
+    volumes:
+      - bcdshare:/etc/bcd
+    links:
+      - "flextesa:sandbox"
+    logging: *bcd-logging
+
+  indexer:
+    restart: always
+    image: ghcr.io/baking-bad/bcdhub-indexer:master
+    environment:
+      - BCD_ENV=sandbox
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=root
+      - SANDBOX_NODE_URI=http://sandbox:20000
+      - SANDBOX_IPFS_GATEWAY=https://cloudflare-ipfs.com
+    depends_on:
+      - db
+      - metrics
+    links:
+      - "flextesa:sandbox"
+    volumes:
+      - bcdshare:/etc/bcd
+    logging: *bcd-logging
+
+  metrics:
+    restart: always
+    image: ghcr.io/baking-bad/bcdhub-metrics:master
+    environment:
+      - BCD_ENV=sandbox
+      - POSTGRES_USER=root
+      - POSTGRES_PASSWORD=root
+      - SANDBOX_NODE_URI=http://sandbox:20000
+      - SANDBOX_IPFS_GATEWAY=https://cloudflare-ipfs.com
+    depends_on:
+      - elastic
+      - db
+    links:
+      - "flextesa:sandbox"
+    volumes:
+      - bcdshare:/etc/bcd
+    logging: *bcd-logging
+
+  flextesa:
+    restart: always
+    image: tqtezos/flextesa:20211206
+    command: hangzbox start
+    environment:
+      - block_time=4
+    ports:
+      - "20000:20000"
+    expose:
+      - 20000/tcp
+    logging: *bcd-logging
+
+  gui:
+    container_name: sandbox-gui
```

Comment:
What about adding a fixed `container_name` for every container? This would avoid some troubles between this file and `sandbox.sh`.

Reply:
Can you clarify?

For reference, the only container we should need to reference by name is the flextesa container. The purpose is so we can use the tezos client (which is already configured correclty in the flextesa container) to deploy the contract, etc.

Reply:
Sorry for this newbie comment, still discovering a lot of things!...

My question was about hard write a `container_name` for every container we use anywhere else. Hence, if I understand correctly, giving a `container_name` to the `flextesa` container should be enough!